WickedEngine/wiNetwork_Linux.cpp

```diff
@@ -1,45 +1,184 @@
 #include "wiPlatform.h"
+#include <cstring>
 
 #ifdef PLATFORM_LINUX
 #include "wiNetwork.h"
 #include "wiBackLog.h"
+#include "wiTimer.h"
+
+#include <sstream>
+#include <string>
+#include <unistd.h>
+#include <errno.h>
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
 
 namespace wiNetwork
 {
-	void Initialize()
+	struct SocketInternal{
+		int handle;
+		~SocketInternal(){
+			int result = close(handle);
+			if(result < 0){
+				assert(0);
+			}
+		}
+	};
+
+	SocketInternal* to_internal(const Socket* param)
 	{
-		wiBackLog::post("TODO wiNetwork_Linux");
+		return static_cast<SocketInternal*>(param->internal_state.get());
 	}
 
-	bool CreateSocket(Socket* sock)
+	void Initialize()
 	{
-		return false;
+		wiTimer timer;
+
+		int result;
+
+		wiBackLog::post("wiNetwork_Linux Initialized (" + std::to_string((int)std::round(timer.elapsed())) + " ms)");
 	}
-	bool Destroy(Socket* sock)
+
+	bool CreateSocket(Socket* sock)
 	{
-		return false;
+		std::shared_ptr<SocketInternal> socketinternal = std::make_shared<SocketInternal>();
+		sock->internal_state = socketinternal;
+
+		socketinternal->handle = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+
+		if(socketinternal->handle == -1){
+			std::stringstream ss;
+			ss << "wiNetwork_Linux error in CreateSocket: Could not create socket";
+			wiBackLog::post(ss.str().c_str());
+			return false;
+		}
+
+		return true;
 	}
 
 	bool Send(const Socket* sock, const Connection* connection, const void* data, size_t dataSize)
 	{
+		if (sock->IsValid()){
+			sockaddr_in target;
+			target.sin_family = AF_INET;
+			target.sin_port = htons(connection->port);
+			std::stringstream ipss;
```

Comment:
This part will use heap allocation (due to `std::string` objects), which is not very good if the Send() will be called frequently. I think the Send() and Receive() functions are probably performance critical in how a networking code might use them.
But this looks like pretty similar to the win32 implementation, where string objects are not used. 
The same for the Receive() function.